File: 01/01.py
from typing import List

class Solution(object):
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        result = []
        map = {}
        for i in range(len(nums)):
            cur = nums[i]
            complement = target-cur
            if complement in map.keys():
                result = [map[complement], i]
            map[cur] = i
        return result

for i in Solution().twoSum([2, 7, 11, 15],9):
    print(i)

File: 01/01.java
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int i = 0; i < nums.length; i++) {
            int cur = nums[i];
            int complement = target - cur;
            if (map.containsKey(complement)) {
                return new int[] { map.get(complement), i };
            }

            map.put(cur, i);
        }
        return new int[0];
    }

    public static void main(String[] args) {
        Solution solution = new Solution();
        int[] result = solution.twoSum(new int[] { 2, 7, 11, 15 }, 9);
        for (int i : result) {
            System.out.print(String.format("%d ", i));
        }
        System.err.println();
    }
}